music/demos/visualizer.html

```diff
@@ -50,21 +51,148 @@ <h1>Visualizers</h1>
+          score in a note by note basis (<b>ScrollType.NOTE</b>) or centering it in bar sized 
+          packed chunks (<b>ScrollType.BAR</b>) every time activeNote is the first one of a
+          bar.
+        </li>
+        <li>
+          Multiple <b>key signature</b> and <b>time signature</b> are supported (but notice 
+          mm.urlToNoteSequence does not translate them from MIDI files yet).
+        </li>
+      </ul>
+    </p>
+    <h3>
+      Mixing it all
+    </h3>
+    In next example you can relize several combined features. Two tracks of a single score have 
+    been filtered in two staffs (left and right hand) which automatically adapt to the appropriate
+    clef according to their note range. Switching Compact visualization mode makes same score be 
+    drawn on different mm.StaffSVGVisualizer instances, from compact mode (like a printed staff), 
+    to proportional mode (placing note heads in x axis similarly to piano-roll rectangles, ideal 
+    for multiple piled up visualizers). Musical symbols have been scaled up according with the 
+    defined 'noteHeight' config value. Visualization mode can be changed during playing, but 
+    given there will be a new instance you can experience a fast rewind before scrolling to the 
+    next active note. Finally G Key has been forced as default key given provided score doesn't 
+    handle any key information.
+    <br>
+    <br>
+
+    <button id="playBtn2" disabled>wait...</button>
+    <span class="input">
+      <b>Compact visualization mode:</b>
+      <input type="checkbox" id="visualization2">
+      <b>Tempo:</b>
+      <input id="tempoInput2" value="100" type="range" min="20" max="240" value="120" step="1">
+      <span id="tempoValue2"></span>
+    </span>
+    <br>
+    <br>
+
+    <div class="visualizer-container" id="container4">
+      <div id="right-hand"></div>
+      <div id="left-hand"></div>
     </div>
-  </section>
+    <br>
+
+    <h3>Multiple Key and Time signatures</h3>
+    Next example adds more notes to the score (+) changing through all keys and the time signatures
+    subset [2/4, 3/4, 4/4] to see how the staff grows and adapts to it. Please notice  mm.Player 
+    does not handle incremental NoteSequence modifications once it has started to play.
+    <br>
+    <br>
+
+    <button id="playBtn3" disabled>wait...</button>
+    <button id="growBtn">+</button>
+    <span class="input">
+      <b>Compact visualization mode:</b>
+      <input type="checkbox" id="visualization3">
+      <b>Tempo:</b>
+      <input id="tempoInput3" value="100" type="range" min="20" max="240" value="120" step="1">
+      <span id="tempoValue3"></span>
+    </span>
+    <br>
+    <br>
+    <div class="visualizer-container" id="container5">
+      <div id="multiple-signatures"></div>
+    </div>
+
+    <h3>Musical symbol catalog and managed use cases</h3>
+    Next score shows all musical symbols (but F clef). In a G clef score, C key and 4/4 time signature, 
```

Comment:
I would include this if possible in the main demo, or not include it at all (I think people expect that notes on this staff visualizer should work as expected, so we don't need to demonstrate this necessarily)

Reply:
There is not an "as expected" musical score standard. There are several interpretations of the different ways of doing (same as you can get form "standards" from different musical SW like Finale, Cubase, Logic...), and not just in the visual point of view, but in the musical rules interpretation, which some of them are subject to composer intention among many choices. This is a way to represent "what you will get visually", which cannot be shown in class comments.

I agree It could get out of the general Visualizers demo and go to a separate demo page.